src/wallet/coinselection.cpp

```diff
@@ -93,19 +93,24 @@ std::optional<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_poo
         bool backtrack = false;
         if (curr_value + curr_available_value < selection_target || // Cannot possibly reach target with the amount remaining in the curr_available_value.
             curr_value > selection_target + cost_of_change || // Selected value is out of range, go back and try other branch
-            (curr_waste > best_waste && (utxo_pool.at(0).fee - utxo_pool.at(0).long_term_fee) > 0)) { // Don't select things which we know will be more wasteful if the waste is increasing
```

Comment:
By removing this criteria for backtracking you significantly expand the search space that will lead a) to longer execution time b) overall less less solutions found due to hard cap of `TOTAL_TRIES`.

Reply:
This doesn't change the search space.   With this extra condition removed, this [code path](https://github.com/bitcoin/bitcoin/blob/master/src/wallet/coinselection.cpp#L98) is used instead.  And in the case where `curr_waste > best_waste` using the code path L98, nothing different happens because of this [condition](https://github.com/bitcoin/bitcoin/blob/master/src/wallet/coinselection.cpp#L104).  Since `best_waste` is better than `curr_waste`, nothing happens and in both branches, `backtrack=true` is set.  Running the benchmarks will show there's no difference.

Reply:
@S3RK While I think the search space is worse and I do think it makes the code more readable.  It could be removed and put into a different PR if you think that would be better.

Reply:
I agree that without limitation to the number of iterations you'd get the same result with your algorithm. But we actually have a hard limit of `TOTAL_TRIES`. That means if there are more combinations of coins than `TOTAL_TRIES` we're not going to explore all of them. This is why we want to exit as early as possible, in particular in this condition allows to cut off branches where we didn't reach the target yet but the solution is already worse our current best.